src/cargo/util/config/mod.rs

```diff
@@ -2215,10 +2215,43 @@ impl Drop for PackageCacheLock<'_> {
     }
 }
 
+#[derive(Debug, Default, Deserialize, PartialEq)]
+#[serde(rename_all = "kebab-case")]
+pub enum CargoHttpProxyAuth {
+    #[default]
+    Auto,
```

Comment:
Does the default of `Auto` introduce any additional HTTP requests for the non-proxy case?

The [curl docs](https://curl.se/libcurl/c/CURLOPT_PROXYAUTH.html) indicate that it might:
> If more than one bit is set, libcurl will first query the site to see what authentication methods it supports and then pick the best one you allow it to use. For some methods, this will induce an extra network round-trip

Reply:
This change focuses on proxy authentication specifically, not authentication to arbitrary HTTP servers. The `Auth` API is used for both problems, thus this paragraph I think. In the non-proxy case there is no additional request to be made.

This could well send an additional request to a proxy that doesn't need authentication, however I don't think it would make a noticeable difference in performance compared to, e.g. the time it takes to update the registry or download kilobytes of crates (?).